lib/post_revisor.rb

```diff
@@ -601,4 +604,27 @@ def successfully_saved_post_and_topic
     @post_successfully_saved && !@topic_changes.errored?
   end
 
+  def revise_likes(deleted_post_ids, acting_user)
+    return if deleted_post_ids.blank?
+
+    already_liked_users = UserAction.where(target_post_id: @post.id, action_type: UserAction::LIKE).pluck(:user_id)
+    new_user_ids_to_like = UserAction.where(target_post_id: deleted_post_ids, action_type: UserAction::LIKE).where.not(user_id: already_liked_users).pluck(:user_id)
+
+    args = {}
+    args[:message] = "Merging posts"
+    args[:take_action] = true
+    args[:flag_topic] = true
+
+    deleted_posts = Post.where(id: deleted_post_ids)
+    new_users_to_like = User.where(id: new_user_ids_to_like)
+
+    new_users_to_like.each do |user|
```

Comment:
Is this right? aren't you calling `remove_act` here on some posts with no actions by that user? 

also dates on the post action are going to be skewed + users are not going to be notified of these new likes.

Reply:
I have pushed code with the new approach, have to test it still though.

Reply:
@SamSaffron can you have a look now?